docs/adr/0006-consensus-governance.md

```diff
@@ -0,0 +1,459 @@
+# ADR 0006: Consensus Governance
+
+## Changelog
+
+- 2020-10-16: Initial draft
+
+## Status
+
+Proposed
+
+## Context
+
+Currently the consensus layer does not contain any on-chain governance
+mechanism so any network upgrades need to be carefully coordinated off-chain.
+An on-chain governance mechanism would allow upgrades to be handled in a more
+controlled (and automatable) manner without introducing the risk of corrupting
+state.
+
+## Decision
+
+This proposal introduces a minimal on-chain governance mechanism where anyone
+can submit governance proposals and the validators can vote where one base unit
```

Comment:
Maybe we can explicitly point out that this includes both self-delegations and delegations from other users, since conceptually sometimes people think about delegation as primarily in the "delegations from other users" case. 

We can just add parentheses at the end: "(including both self-delegations and delegations from other users)."